aten/src/ATen/native/DilatedMaxPool2d.cpp

```diff
@@ -58,23 +50,58 @@ void max_pool2d_with_indices_out_cpu_template(
   const int64_t outputHeight = pooling_output_shape<int64_t>(inputHeight, kH, padH, dH, dilationH, ceil_mode);
   const int64_t outputWidth = pooling_output_shape<int64_t>(inputWidth, kW, padW, dW, dilationW, ceil_mode);
 
+  const auto memory_format = input.suggest_memory_format();
   pool2d_shape_check(
     input,
     kH, kW, dH, dW, padH, padW, dilationH, dilationW,
     nInputPlane,
     inputHeight, inputWidth,
-    outputHeight, outputWidth, input.suggest_memory_format());
+    outputHeight, outputWidth, memory_format);
 
   /* resize output and indices */
   if (input.ndimension() == 3) {
-    output.resize_({nInputPlane, outputHeight, outputWidth});
+    set_output(0, {nInputPlane, outputHeight, outputWidth}, {}, input.options().memory_format(memory_format), input.names());
```

Comment:
@ezyang Why there must use `input.options().memory_format(memory_format)`？When I use `input.options()` instead, the `nhwc testcase` will be wrong.   
In my idea, `input.options()` will copy the `input`'s option(including `memory_format`) to `output`. Why does this not happen?

Reply:
It's a very goofy problem: `options()` was grandfathered to NOT tell you the memory format. So it never records memory format info. At some point we should fix this...